### PR TITLE
Implement SDL-0177 Alert Icon

### LINKED ISSUE #995: [SDL 0177] Alert icon
Proposal: [Alert icon](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0177-alert-icon.md)

> This proposal is about adding the RPC `alert` to include an icon.

Review: smartdevicelink/sdl_evolution#515

### PR DESCRIPTION
**Pull request by joeljfischer**
Fixes #995 

This PR **ready** for review.

### Risk
This PR makes **minor** API changes.

### Testing Plan
Unit tests and smoke tests

### Summary
Implements [SDL-0177 Alert Icon](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0177-alert-icon.md).

##### Enhancements
* Add an `Image` to `Alert`, allowing setting an image in an `Alert` on supported systems.

### Tasks Remaining:
- [x] Test against Core PR when available.

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: Example Apps/Example ObjC/ButtonManager.m [hunk 1]
```diff
@@ -73,14 +73,18 @@ - (void)setToggleEnabled:(BOOL)toggleEnabled {
...
     SDLSoftButtonState *alertTextState = [[SDLSoftButtonState alloc] initWithStateName:AlertSoftButtonTextState text:AlertSoftButtonText image:nil];
 
     __weak typeof(self) weakself = self;
     SDLSoftButtonObject *alertSoftButton = [[SDLSoftButtonObject alloc] initWithName:AlertSoftButton states:@[alertImageAndTextState, alertTextState] initialStateName:alertImageAndTextState.name handler:^(SDLOnButtonPress * _Nullable buttonPress, SDLOnButtonEvent * _Nullable buttonEvent) {
         if (buttonPress == nil) { return; }
 
-        [weakself.sdlManager sendRequest:[AlertManager alertWithMessageAndCloseButton:@"You pushed the soft button!" textField2:nil]];
+        [weakself.sdlManager.fileManager uploadArtwork:[SDLArtwork artworkWithImage:[UIImage imageNamed:CarBWIconImageName] asImageFormat:SDLArtworkImageFormatPNG] completionHandler:^(BOOL success, NSString * _Nonnull artworkName, NSUInteger bytesAvailable, NSError * _Nullable error) {
```
Comment (justingluck93) [inline Example Apps/Example ObjC/ButtonManager.m]: Just a question, is there a reason the image is being uploaded here again? Is the image uploaded when the soft button state is created? Is it just to make sure the file is on the system before using it?

File: SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLAlertSpec.m [hunk 2]
```diff
@@ -9,99 +9,320 @@
...
+            expect(testAlert.alertText3).to(beNil());
+            expect(testAlert.ttsChunks).to(haveCount(1));
+            expect(testAlert.duration).to(equal(defaultDuration));
+            expect(testAlert.playTone).to(equal(testPlayTone));
+            expect(testAlert.progressIndicator).to(beFalse());
+            expect(testAlert.softButtons).to(beNil());
+            expect(testAlert.alertIcon).to(beNil());
+        });
+
+        it(@"should initialize correctly with initWithTTSChunks:alertText1:alertText2:alertText3:playTone:duration:softButtons:", ^{
```
Comment (justingluck93) [inline SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLAlertSpec.m]:
```suggestion
        it(@"should initialize correctly with initWithAlertText1:alertText2:alertText3:ttsChunks:playTone:progressIndicator:duration:softButtons:alertIcon:", ^{
```

File: SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLAlertSpec.m [hunk 3]
```diff
@@ -9,99 +9,320 @@
...
+            expect(testRequest.ttsChunks).to(beNil());
+            expect(testRequest.duration).to(beNil());
+            expect(testRequest.playTone).to(beNil());
+            expect(testRequest.progressIndicator).to(beNil());
+            expect(testRequest.softButtons).to(beNil());
+        });
     });
 });
 
+
```
Comment (justingluck93) [inline SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLAlertSpec.m]: Remove extra line

File: Example Apps/Example ObjC/AlertManager.m [hunk 4]
```diff
@@ -31,8 +32,8 @@ + (SDLAlert *)alertWithMessage:(NSString *)textField1 textField2:(nullable NSStr
  *  @param textField2  The second line of a message to display in the alert
  *  @return            An SDLAlert object
  */
-+ (SDLAlert *)alertWithMessageAndCloseButton:(NSString *)textField1 textField2:(nullable NSString *)textField2 {
-    return [[SDLAlert alloc] initWithAlertText1:textField1 alertText2:textField2 alertText3:nil duration:5000 softButtons:@[[self sdlex_okSoftButton]]];
++ (SDLAlert *)alertWithMessageAndCloseButton:(NSString *)textField1 textField2:(nullable NSString *)textField2 iconName:(nullable NSString *)iconName {
```
[the remark this answers is not in the document]
Comment (joeljfischer) [reply Example Apps/Example ObjC/AlertManager.m]: fixed

**Pull request merged**